client/src/services/gameHelper.js

```diff
@@ -454,6 +454,33 @@ class GameHelper {
     return game.galaxy.stars.sort((a, b) => b.location.y - a.location.y)[0].location.y
   }
 
+  getSpecialistModifierValue(starSpecialists, specialistId, modifierType, modifierName) {
+    const specialist = starSpecialists.find(x => x.id === specialistId)
+    if (specialist && specialist.modifiers && specialist.modifiers[modifierType]) {
+      return specialist.modifiers[modifierType][modifierName] || 1;
+    }
+    return 1;
```

Comment:
This function is trying to do too much, not all specialist modifier values will default to 1 if they aren't present. You're probably better off explicitly searching for the modifier you care about in the `getEconomyUpgradeForStar` function instead of a clever all encompassing function that handles every possible scenario.